### ISSUE TITLE
what about giving names to shared variables in updates?

### ISSUE THREAD

**Issue by kmyid** [Lasagne/Lasagne#600]
Hi,

Wouldn't it be a good idea to give names to theano shared variables in the update functions?
I am writing a save network function to store things in hdf5 files and thought it would be a good idea to have names for shared variables in updates.
In this case, we can store the state of the optimizers easily using the returned updates dictionary to recover later on, in case there is the need (e.g. continuing after a crash).
It wouldn't be hard as you would simply need to append somthing at the end of the parameter names.
E.g. in case of momentum velocity can be layer0.W.velocity

What do you guys think?

Kwang

**Comment by f0k** [comment]
Yes, this was recently proposed on the mailing list as well: https://groups.google.com/forum/#!topic/lasagne-users/MsUGPlGLvuk
We'd welcome a PR, we'll just likely have to discuss the naming scheme. Would you like to do a PR? Matthew said he'd do one, but it'd take some time -- feel free to take the task yourself!

**Comment by kmyid** [comment]
> Matthew said he'd do one, but it'd take some time

Same here actually, if Matthew didn't do a PR by the time I get some free time, I'll create one :-)

**Comment by f0k** [comment]
Moved to v0.3. Would be nice to have, but is not critical for the release.
